### ISSUE TITLE
line shader compilation failed when using GLES

### ISSUE THREAD

**Issue by thewierdnut** [endless-sky/endless-sky#10520]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Getting the "Shader compilation failed." popup when building with GLES. 

### Steps to Reproduce

1. Build with GLES: (I used `cmake .. -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=0 -DES_USE_VCPKG=0 -DES_GLES=1`)
2. Attempt to run.

### Expected Behavior

Program starts.

### Screenshots

```
#version 320 es
// vertex line shader
uniform vec2 scale;
uniform vec2 start;
uniform vec2 end;
uniform float width;
uniform int cap;
in vec2 vert;
out vec2 pos;
void main() {
    vec2 unit = normalize(end - start);
    vec2 origin = vert.y > 0.0 ? start : end;
    float widthOffset = width + 1;
    float capOffset = (cap == 1) ? widthOffset : 1;
    pos = origin + vec2(unit.y, -unit.x) * vert.x * widthOffset - unit * capOffset * vert.y;
    gl_Position = vec4(pos / scale, 0, 1);
    gl_Position.y = -gl_Position.y;
    gl_Position.xy *= 2.0;
}
0:13(22): error: could not implicitly convert operands to arithmetic operator
0:14(33): warning: `widthOffset' used uninitialized
0:14(33): error: second and third operands of ?: operator must have matching types
0:15(50): warning: `widthOffset' used uninitialized
0:15(71): warning: `capOffset' used uninitialized

Shader compilation failed.
```

### Operating System

Debian bullseye

### Game Source

Built from source

### Game Version

v0.10.9

### Additional Information

```
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: AMD (0x1002)
    Device: AMD Radeon Graphics (gfx1036, LLVM 15.0.6, DRM 3.49, 6.1.0-22-amd64) (0x164e)
    Version: 22.3.6
    Accelerated: yes
    Video memory: 512MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
```

**Comment by thewierdnut** [comment]
If I explicitly set the constant `1` in those float operations to `1.0`, then those errors go away, and I get two new errors during the linking stage: 

```
error: declarations for uniform `start` have mismatching precision qualifiers
error: declarations for uniform `cap` have mismatching precision qualifiers
```

If I add 
```glsl
	precision mediump float;
	precision mediump int;
```
To both shaders, then it finally compiles and starts.

**Comment by thewierdnut** [comment]
An additional note... Using medium precision for float results in glitchy scrollbars when the scrollbar viewable area is a high percentage of the total. High precision solves that problem, but gles does require high precision to be available (though in practice, I have never seen it absent)

I believe that all the major desktop opengl vendors use high precision regardless of the shader precision specifier, so you won't be able to see this glitching on a desktop compile.

**Comment by thewierdnut** [comment]
Here is a screenshot showing the scrollbar as it appears with `precision mediump float;`. It is a tiny dot, whereas if I set `precision highp float;`, that bar would extend from that dot to the top of the screen. The scrollbar appears to be passing the same arguments to the shader either way. 

![image](https://github.com/user-attachments/assets/15004217-e8ac-41e9-b933-c51bf117eae1)

**Comment by tibetiroka** [comment]
Does this still happen if you revert to 6c0301c82c1cf5f595824327cfea8ba9af75a927?

**Comment by tibetiroka** [comment]
@Koranir Any thoughts? You are rather familiar with shaders in general, and also with the new changes.

**Comment by thewierdnut** [comment]
No, 6c0301c is fine, but 12b9e9ec fails.

**Comment by Koranir** [comment]
I'll investigate this further.

**Comment by Koranir** [comment]
> An additional note... Using medium precision for float results in glitchy scrollbars when the scrollbar viewable area is a high percentage of the total. High precision solves that problem, but gles does not require high precision to be available (though in practice, I have never seen it absent)

The OpenGL ES 3.0 standard requires that high precision is always available, but ES 2.0 does not. As far as I'm aware, we target OpenGL ES 3.0 with Endless Sky.
